docs-website/generateDocsDir.ts

```diff
@@ -158,6 +158,13 @@ function markdown_guess_title(
   } else {
     // Find first h1 header and use it as the title.
     const headers = contents.content.match(/^# (.+)$/gm);
+
+    if (!headers) {
+      throw new Error(
+        `${filepath} must have at least one h1 header for setting the title`
```

Comment:
it needs exactly one h1 header

Reply:
what happens if there's more than one?

Reply:
There's a check for that below - it will throw an error as well. Mainly just wanted to make the error message more clear

Reply:
oh weird – I have two h1's in our bigquery and snowflake source docs (for main + usage stats) and it seems to compile without issue. seems it also checks for no code blocks via `&& contents.content.indexOf("```") < 0`?